core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -720,7 +720,7 @@ private[spark] class MapOutputTrackerMaster(
     }
   }
 
-  def registerMergeResult(shuffleId: Int, reduceId: Int, status: MergeStatus) {
+  def registerMergeResult(shuffleId: Int, reduceId: Int, status: MergeStatus): Unit = {
```

Comment:
Note to reviewers: this was added as part of push-based shuffle so just fixing the below warning:
```
[warn] /home/runner/work/spark/spark/core/src/main/scala/org/apache/spark/MapOutputTracker.scala:723:79: [deprecation @  | origin= | version=2.13.0] procedure syntax is deprecated: instead, add `: Unit =` to explicitly declare `registerMergeResult`'s return type
[warn]   def registerMergeResult(shuffleId: Int, reduceId: Int, status: MergeStatus) {
```